src/api/spec/controllers/person/notifications_controller_spec.rb

```diff
@@ -94,4 +94,46 @@
       it { expect(response).to have_http_status(:unauthorized) }
     end
   end
+
+  describe '#update' do
+    let!(:notification) { create(:web_notification, :comment_for_package, subscriber: user) }
+
+    before do
+      Flipper[:notifications_redesign].enable
+    end
+
+    context 'called by an unauthorized user' do
+      let(:other_user) { create(:confirmed_user, :in_beta) }
+
+      before do
+        login other_user
+        put :update, params: { format: :xml, id: notification.id }
+      end
+
+      it { expect(response).to have_http_status(:forbidden) }
+    end
+
+    context 'called by an authorized user' do
+      before do
+        login user
+        put :update, params: { format: :xml, id: notification.id }
+      end
+
+      it 'toggles the delivered attribute' do
+        notification.reload
```

Comment:
Why is doing `reload` twice needed?

Reply:
I had some weird problems without the reload, nevermind it's not needed!